Combinator/Combinator.fs

```diff
@@ -169,14 +169,26 @@ module Combinator =
     let manyTill p pEnd = manyTillWithCount 0 p pEnd
 
     let manyTill1 p pEnd = manyTillWithCount 1 p pEnd
-        
-    let manyN num (parser) =                              
-        takeTillWithCount 0 (fun i -> List.length i > num) parser    >>= fun result -> 
-            if result.Length <> num then
-                failwith ("Error, attempted to match " + num.ToString() + " but only got " + result.Length.ToString())
 
-            preturn result          
-    
+
+
+    let manyN num parser state =
+        let didFind found currentState =
```

Comment:
Is this function necessary? It just returns the list reversed and returns the current state.  Seems like we can update the final clause to just do that inline and remove the extra code